### PR TITLE
add VIA support for hiney PCBs

### PR DESCRIPTION
**Pull request by hineybush**
## Description

Add VIA keymaps for: h87a, h88, hbcp, h75 Singa V3.

The RGB lighting still does not work 100% - just getting the VIA keymaps in.

Add bootmagic lite for h88, PhysiX and SM68 PCBs. VIA keymaps coming soon for the latter two.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Wilba6582** [comment]
👍 

The RGB lighting will just magically work after I do the next PR of VIA changes. Promise.

File: keyboards/hineybush/h87a/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+void matrix_scan_user(void) {
+
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  return true;
+}
+
+void led_init_ports(void) {
+  DDRD |= (1<<5); // OUT
```
Comment (drashna) [inline keyboards/hineybush/h87a/keymaps/via/keymap.c]:
```suggestion
    setPinOutput(D5);
```

File: keyboards/hineybush/h87a/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  DDRE |= (1<<6); // OUT
```
Comment (drashna) [inline keyboards/hineybush/h87a/keymaps/via/keymap.c]:
```suggestion
    setPinOutput(E6);
```

File: keyboards/hineybush/h87a/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    DDRD &= ~(1 << 5); PORTD &= ~(1 << 5);
+  }
+
+  if (usb_led & (1 << USB_LED_SCROLL_LOCK)) {
+    DDRE |= (1 << 6); PORTE &= ~(1 << 6);
+  } else {
+    DDRE &= ~(1 << 6); PORTE &= ~(1 << 6);
+  }
+
+}
```
Comment (drashna) [inline keyboards/hineybush/h87a/keymaps/via/keymap.c]: This should be replaced with: 
```c
bool led_update_user(led_t led_state) {
    writePin(D5, !led_state.caps_lock);
    writePin(E6, !led_state.scroll_lock);
    return true;
}
```

File: keyboards/hineybush/h87a/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+void matrix_scan_user(void) {
+
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  return true;
+}
```
Comment (fauxpark) [inline keyboards/hineybush/h87a/keymaps/via/keymap.c]: These functions can be deleted.

File: keyboards/hineybush/h88/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  return true;
+}
```
Comment (fauxpark) [inline keyboards/hineybush/h88/keymaps/via/keymap.c]: As above.

File: keyboards/hineybush/hbcp/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  return true;
+}
+
+void matrix_init_user(void) {
+
+}
+
+void matrix_scan_user(void) {
+
+}
```
Comment (fauxpark) [inline keyboards/hineybush/hbcp/keymaps/via/keymap.c]: As above.

File: keyboards/hineybush/hbcp/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+  QMKBEST = SAFE_RANGE,
+  QMKURL
+};
```
Comment (fauxpark) [inline keyboards/hineybush/hbcp/keymaps/via/keymap.c]: As above.

**Pull request merged**